cvat/apps/engine/views.py

```diff
@@ -61,809 +79,1122 @@
+			params.append(Parameter(name="tf-image", value=tf_image))
+			params.append(Parameter(name="sys-node-pool", value=machine))
+			if 'TFRecord' in form_data['dump_format']:
+				if form_data['base_model'] and "tfod" in form_data['base_model']:
+					ref_model_path = os.getenv('AWS_S3_PREFIX')+'/'+os.getenv('ONEPANEL_RESOURCE_NAMESPACE')+'/'+form_data['base_model']
+				else:
+					ref_model_path = "base-models/"+form_data['ref_model']
+				slogger.glob.info("TF ref model path {}".format(ref_model_path))
+				params.append(Parameter(name='model-path',value=os.getenv('AWS_S3_PREFIX')+'/'+os.getenv('ONEPANEL_RESOURCE_NAMESPACE')+'/'+os.getenv('ONEPANEL_RESOURCE_UID')+'/models/'+db_task.name+"_tfod_"+form_data['ref_model']+'_'+stamp+'/'))
+				params.append(Parameter(name='ref-model-path', value=ref_model_path))
+				params.append(Parameter(name='num-classes', value=str(num_classes)))
+				params.append(Parameter(name="ref-model", value=form_data['ref_model']))
+				body = onepanel.core.api.CreateWorkflowExecutionBody(parameters=params,
+				workflow_template_uid = os.getenv('ONEPANEL_OD_TEMPLATE_ID')) 
+			else:
+				if form_data['base_model'] and "maskrcnn" in form_data['base_model']:
+					ref_model_path = os.getenv('AWS_S3_PREFIX')+'/'+os.getenv('ONEPANEL_RESOURCE_NAMESPACE')+'/'+form_data['base_model']
+				else:
+					ref_model_path = "base-models/maskrcnn"
+				slogger.glob.info("maskrcnn ref model path {}".format(ref_model_path))
+				params.append(Parameter(name='model-path',value=os.getenv('AWS_S3_PREFIX')+'/'+os.getenv('ONEPANEL_RESOURCE_NAMESPACE')+'/'+os.getenv('ONEPANEL_RESOURCE_UID')+'/models/'+db_task.name+"_maskrcnn_"+stamp+'/'))
+				params.append(Parameter(name='ref-model-path', value=ref_model_path))
+				params.append(Parameter(name='num-classes', value=str(num_classes+1)))
+				params.append(Parameter(name='stage-1-epochs', value=str(args_and_vals['--stage1_epochs'])))
+				params.append(Parameter(name='stage-2-epochs', value=str(args_and_vals['--stage2_epochs'])))
+				params.append(Parameter(name='stage-3-epochs', value=str(args_and_vals['--stage3_epochs'])))
+				body = onepanel.core.api.CreateWorkflowExecutionBody(parameters=params,
+				workflow_template_uid = os.getenv('ONEPANEL_MASKRCNN_TEMPLATE_ID')) 
+			try:
+				api_response = api_instance.create_workflow_execution(namespace, body)
+				return Response(data="Workflow executed", status=status.HTTP_200_OK)
+			except ApiException as e:
+				slogger.glob.exception("Exception when calling WorkflowServiceApi->create_workflow_execution: {}\n".format(e))
+				return Response(data="error occured", status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+		return Response(data=20, status=status.HTTP_200_OK)
+
+
+	# @permission_required(perm=['engine.task.change'],
+	# fn=objectgetter(TaskModel, 'tid'), raise_exception=True)
+	@action(detail=True, methods=['POST'], url_path='tracking')
+	def tracking(self, request, pk):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5609236)